viewport-scaling.html

```diff
@@ -0,0 +1,222 @@
+        </p>
+      </details>
+    </header>
+    <main style='text-align: center;'>
+      <p>Click 'Enter XR' to see content</p> 
+    </main>
+    <script type="module">
+      import {WebXRButton} from './js/util/webxr-button.js';
+      import {Scene, WebXRView} from './js/render/scenes/scene.js';
+      import {Renderer, createWebGLContext} from './js/render/core/renderer.js';
+      import {Gltf2Node} from './js/render/nodes/gltf2.js';
+      import {SkyboxNode} from './js/render/nodes/skybox.js';
+      import {QueryArgs} from './js/util/query-args.js';
+
+      import WebXRPolyfill from './js/third-party/webxr-polyfill/build/webxr-polyfill.module.js';
+      if (QueryArgs.getBool('usePolyfill', true)) {
+        let polyfill = new WebXRPolyfill();
+      }
+
+      // XR globals.
+      let xrButton = null;
+      let xrRefSpace = null;
+
+      // WebGL scene globals.
+      let gl = null;
+      let renderer = null;
+      let scene = new Scene();
+      let scale = 1;
+      let scale_down = true;
+      let useLayerCheckbox = document.getElementById('useLayer');
+      let useLayer = false;
+      // layer support
+      let xrGLFactory = null;
+      let xrFramebuffer = null;
+      let proj_layer = null;
+      let count = true;
+
+      scene.addNode(new Gltf2Node({url: 'media/gltf/space/space.gltf'}));
+      scene.addNode(new SkyboxNode({url: 'media/textures/milky-way-4k.png'}));
+
+      function initXR() {
+        xrButton = new WebXRButton({
+          onRequestSession: onRequestSession,
+          onEndSession: onEndSession
+        });
+        document.querySelector('header').appendChild(xrButton.domElement);
+
+        if (navigator.xr) {
+          navigator.xr.isSessionSupported('immersive-vr').then((supported) => {
+            xrButton.enabled = supported;
+          });
+        }
+      }
+
+      function onRequestSession() {
+        useLayer = useLayerCheckbox.checked;
+        return navigator.xr.requestSession('immersive-vr', {
+          requiredFeatures: useLayer?['layers']:[],
+        }).then(onSessionStarted);
+      }
+
+      function onSessionStarted(session) {
+        xrButton.setSession(session);
+
+        session.addEventListener('end', onSessionEnded);
+
+        gl = createWebGLContext({
+          xrCompatible: true, webgl2: true
+        });
+
+        renderer = new Renderer(gl);
+
+        scene.setRenderer(renderer);
+
+        if (useLayer) {
+            xrFramebuffer = gl.createFramebuffer();
+            xrGLFactory = new XRWebGLBinding(session, gl);            
+        } else {
+            let gllayer = new XRWebGLLayer(session, gl);
+            gllayer.fixedFoveation = 1.0;
+            session.updateRenderState({ baseLayer: gllayer });
+        }
+
+        session.requestReferenceSpace('local').then((refSpace) => {
+          xrRefSpace = refSpace;
+
+          if (useLayer) {
+            proj_layer = xrGLFactory.createProjectionLayer({ space: refSpace, stencil: false });
```

Comment:
Projection Layers are the only layer type you can create without providing a space, so passing refSpace here is ignored.

If you want you could move this up to happen just after creating the binding, which would make the two paths a little more similar, but I don't think it's critical.